pandas/_libs/tslibs/offsets.pyx

```diff
@@ -458,7 +458,17 @@ cdef class BaseOffset:
             FutureWarning,
             stacklevel=1,
         )
-        return self.apply(other)
+        return self._apply(other)
+
+    def apply(self, other):
```

Comment:
can you type & add a doc-string (followup ok)

Reply:
will add to my next CLN branch